### PR TITLE
Implementation of #23 Stropping and #62 Add globals

### PR DESCRIPTION
**Pull request by thirtytwobits**
This is a complete PR for the Stropping implementation and the library
access to add globals with some other work on #64 mixed in. It is not
a 1.1.3 release candidate until #64 is complete.

### CODE REVIEW

File: src/nunavut/jinja/lang/__init__.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def get_supported_languages() -> typing.Iterable[str]:
     return __language_modules__.keys()
 
 
-def add_language_support(language_name: str, environment: Environment) -> None:
+def add_language_support(language_name: str, environment: Environment, make_implicit: bool = False) -> None:
```
Comment (pavel-kirienko) [inline src/nunavut/jinja/lang/__init__.py]: Missing docs for `make_implicit`

File: src/nunavut/jinja/lang/c.py [hunk 2]
```diff
@@ -16,6 +16,153 @@
...
+    '_Alignof',
+    '_Atomic',
+    '_Bool',
+    '_Complex',
+    '_Generic',
+    '_Imaginary',
+    '_Noreturn',
+    '_Pragma',
+    '_Static_assert',
+    '_Thread_local'
```
Comment (pavel-kirienko) [inline src/nunavut/jinja/lang/c.py]: The following are all reserved:

- `_[A-Z_]+[a-zA-Z0-9_]*`
- `(is|to|str|mem|wcs|atomic_|cnd_|mtx_|thrd_|tss_|memory_)[a-z][a-zA-Z_0-9]*`
- `u?int[a-zA-Z_0-9]*_t`
- `E[A-Z0-9][a-zA-Z_0-9]*`
- `FE[A-Z][a-zA-Z_0-9]*`
- `U?INT[a-zA-Z_0-9]*_(MAX|MIN|C)`
- `(PRI|SCN)[a-zX][a-zA-Z_0-9]*`
- `LC_[A-Z][a-zA-Z_0-9]*`
- `SIG_?[A-Z][a-zA-Z_0-9]*`
- `TIME_[A-Z][a-zA-Z_0-9]*`
- `ATOMIC_[A-Z][a-zA-Z_0-9]*`

https://en.cppreference.com/w/c/language/identifier
https://en.cppreference.com/w/cpp/language/identifiers

Would it not be better to make this a list of regexps?

File: src/nunavut/jinja/lang/__init__.py (hunk 1 above)
Comment (thirtytwobits) [reply src/nunavut/jinja/lang/__init__.py]: Oh, good catch. I also forgot to test for a double call to this.

File: src/nunavut/jinja/lang/c.py (hunk 2 above)
Comment (thirtytwobits) [reply src/nunavut/jinja/lang/c.py]: > Would it not be better to make this a list of regexps?

It would now! Good catch. Thanks.

**Comment by pavel-kirienko** [comment]
`_` as a prefix won't work for Python because by convention it affects the member visibility. In Python, underscores should be appended instead.

File: src/nunavut/jinja/lang/c.py (hunk 2 above)
Comment (thirtytwobits) [reply src/nunavut/jinja/lang/c.py]: > (is|to|str|mem|wcs|atomic_|cnd_|mtx_|thrd_|tss_|memory_)[a-z][a-zA-Z_0-9]*

My reading of cppreference is that this pattern is actually `(is|to|str|mem|wcs|atomic_|cnd_|mtx_|thrd_|tss_|memory_)[a-z]+`. Am I missing something?

Comment (pavel-kirienko) [reply src/nunavut/jinja/lang/c.py]: It says "followed by a lowercase letter", not "by lowercase letters". Although, now that I'm thinking of it, it doesn't make much sense; perhaps they meant the latter but worded the description poorly (or I am simply misreading it). I tried looking at the standard but couldn't immediately find the relevant sections. Let's go with your interpretation.

Comment (thirtytwobits) [reply src/nunavut/jinja/lang/c.py]: I'll change to `^(is|to|str|mem|wcs|atomic_|cnd_|mtx_|thrd_|tss_|memory_|memory_order_)[a-z]{1}` which is more correct and probably more efficient.

**Pull request closed without merging**